bpf/lib/maps.h

```diff
@@ -284,3 +285,13 @@ tail_call_internal(struct __ctx_buff *ctx, const __u32 index, __s8 *ext_err)
 	return DROP_MISSED_TAIL_CALL;
 }
 #endif /* SKIP_CALLS_MAP */
+
+#ifdef ENABLE_CPU_MAP
+struct {
+    __uint(type, BPF_MAP_TYPE_CPUMAP);
+    __type(key, __u32);
+    __type(value, struct bpf_cpumap_val);
+    __uint(pinning, LIBBPF_PIN_BY_NAME);
```

Comment:
This is dangerous if CPU maps ever contain program pointers for multi-xdp. Loading multiple ELFs (1 per xdp device) will start clobbering entries in this shared map and lead to a split-brain where packets entering one interface will end up executing programs belonging to another.

See my other comments, there's little value in pinning this, and not doing so will save us 500 lines of boilerplate.

Reply:
To make sure I understand, you're referencing a use case where we use the CPU MAP in the future to implement multi-xdp for different use cases, and those use cases collide with one another since the map is shared?